www/src/public/js/controllers/builder/main.js

```diff
@@ -2070,6 +2308,332 @@
+                        runeStats.rent += 360;
+                        runeStats.rangedAccuracy += 2;
+                        runeStats.charmName = runeStats.charmName + "Perthro/";
+                        break;
+                    case "R":
+                        //-3 ac
+                        runeStats.rent += 135;
+                        runeStats.ac -= 3;
+                        runeStats.charmName = runeStats.charmName + "Thurisaz/";
+                        break;
+                    case "S":
+                        //2 spell crit
+                        runeStats.rent += 103;
+                        runeStats.spellcrit += 2;
+                        runeStats.charmName = runeStats.charmName + "Hagalaz/";
+                        break;
+                    case "T":
+                        //2 spell dam
+                        runeStats.rent += 675;
+                        runeStats.spelldam += 2;
+                        runeStats.charmName = runeStats.charmName + "Nauthiz/";
+                        break;
+                    case "U":
+                        //2 mana redux
+                        runeStats.rent += 203;
+                        runeStats.concentration++;
+                        runeStats.manaReduction += 2;
+                        runeStats.charmName = runeStats.charmName + "Sowilo/";
+                        break;
+                    case "V":
+                        //detect invis
+                        runeStats.rent += 900;
+                        runeStats.charmName = runeStats.charmName + "Jera/";
+                        break;
+                    case "W":
+                        //see dark
+                        runeStats.rent += 900;
+                        runeStats.charmName = runeStats.charmName + "Dagaz/";
+                        break;
+                    case "X":
+                        //detect illusion
+                        runeStats.rent += 900;
+                        runeStats.charmName = runeStats.charmName + "Othala/";
+                        break;
+                    case "Y":
+                        //sneak
+                        runeStats.charmName = runeStats.charmName + "Ingwaz/";
+                        break;
+                }
+            }
+
+            return runeStats;
+        };
+        
+        /** swaps isRuneCrafting for modal hiding purposes **/
+        $scope.runeCraftButtonClick = function() {
+               $scope.isRuneCrafting = !$scope.isRuneCrafting;
+               console.log("Button click");
+               $scope.getCurrentRunes($scope.currentItemIndex);
+   
+        };
+
+        $scope.getCurrentRunes = function(index) {
+           console.log("Get runes:", index);
+
+           this.charm1Selector = 'A';
+           this.charm2Selector = 'A';
+           this.charm3Selector = 'A';
+           this.charm4Selector = 'A';
+           this.charm5Selector = 'A';
+
+            switch (index) {
```

Comment:
This could probably use the same treatment from the `$scope.runeCraftCharms` function to reduce this code by quite a bit.

Reply:
Sounds good, make sure to assign issue #139 to yourself